python/ray/util/multiprocessing/pool.py

```diff
@@ -243,6 +257,13 @@ def run(self):
                 # instead of on a list encaspulating that result
                 self._callback(aggregated_batch_results[0])
 
+    def stop(self):
+        # Call this to interrupt the run function.
+        # The run() will terminate when all pending jobs have completed.
+        self._stop_event.set()
```

Comment:
I believe the `_stop_event` should be [`clear`ed](https://docs.python.org/3/library/threading.html#threading.Event.clear) at some point. Otherwise, `_stop_event` will be permanently set after `stop` is called once.

Reply:
Thanks for checking that. As far as I could tell, the `ResultThread` object was only designed to be run-once in the original implementation. It would get an `object_refs` array or a number (`total_object_refs`) and collect exactly that number of objects before terminating the `run()` loop. I don't think that it provides for a clean restart ability. Hence my focus on only a graceful shutdown. 

Perhaps `shutdown` is a better name than `stop` then?

Reply:
Ah, good idea. I think `shutdown` is better than `stop` in this case.

Reply:
done!